### DISCUSSION TITLE
How does the velocity curve in schism work?

### DISCUSSION THREAD

**Discussion by Corsaka** [schismtracker/schismtracker#446] [Q&A] [answered]
I've been attempting to transfer a module file to a modern DAW in order to explain it to people with an interface they're used to, but I can't seem to nail how the velocity functions work.

I've examined the code that I assumed was [responsible](https://github.com/schismtracker/schismtracker/blob/master/player/sndmix.c), but the only thing that looks like it might be relevant has 128 entries instead of the 65 levels that the tracker uses. How does the "global volume"/"default volume" inside of the tracker translate to MIDI velocity?

**Comment by sagamusix** [comment, marked as answer]
All volumes in Impulse Tracker / Schism Tracker are linear, and are simply multiplied with each other. So a note at v64, with global volume 128, channel volume 64 will sound at full volume, and if you halve either of those numbers, the note will be half as loud, for example.

MIDI velocity does not have a defined volume curve, many synths have custom velocity mapping curves (sometimes even different for each individual layer of a patch). But in general, the "default" volume behaviour in the world of MIDI is logarithmic. The table that you found is only used for Schism's MIDI output. Samples don't use it.

**Comment by sagamusix** [comment]
@Corsaka did this answer your question? If so, please close the issue, otherwise specify which information is missing.

**Comment by Corsaka** [comment]
sorry yes this was helpful, the issue lies with the samples i've ripped from the tracker file rather than schism - thanks for the reply!
